### ISSUE TITLE
step_select_boruta and step_select_mrmr need method for internally handling NAs

### ISSUE THREAD

**Issue by kransom14** [stevenpawley/recipeselectors#11]
step_select_boruta and step_select_mrmr cannot handle data with missing/NA values. This requires the user to remove or impute NAs in a recipe step prior to the feature selection step in order to use these feature selection steps, which might not be desirable. It would be handy if step_select_boruta and step_select_mrmr could internally omit NAs which would allow the user to preserve them in the training data.

**Comment by stevenpawley** [comment]
Is there a reason why adding a `step_impute_` step before the filter-based step is undesirable? I guess it you are specifically wanting your model to handle the missing values, e.g., if using XGBoost, then you might not want NAs imputed by another method? However, overall, most steps in the 'recipes' package do not handle NAs, and given the diversity of imputation methods, I'm not sure if just adding approaches on a recipe step basis is the way to go, given the composable style of tidymodels.

You could add a 'missing' column first using `step_indicate_na`, impute the NAs and them potentially add them back in if you wanted the model at the end of the pipeline to handle them using its own method.

**Comment by kransom14** [comment]
Yes, it would be for development of a model capable of handling the NAs, e.g. XGBoost.

From: Steven Pawley ***@***.***>
Sent: Friday, November 11, 2022 8:31 AM
To: stevenpawley/recipeselectors ***@***.***>
Cc: Ransom, Katherine M ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [stevenpawley/recipeselectors] step_select_boruta and step_select_mrmr need method for internally handling NAs (Issue #11)




 This email has been received from outside of DOI - Use caution before clicking on links, opening attachments, or responding.



Is there a reason why adding a step_impute_ step before the filter-based step is undesirable? I guess it you are specifically wanting your model to handle the missing values, e.g., if using XGBoost, then you might not want NAs imputed by another method? However, overall, most steps in the 'recipes' package do not handle NAs, and given the diversity of imputation methods, I'm not sure if just adding approaches on a recipe step basis is the way to go, given the composable style of tidymodels.

You could add a 'missing' column first using step_indicate_na, impute the NAs and them potentially add them back in if you wanted the model at the end of the pipeline to handle them using its own method.

-
